core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -99,6 +99,10 @@ class SparkEnv (
 
   private[spark] var executorBackend: Option[ExecutorBackend] = None
 
+  private[spark] var cachedArrowBatchServerPort: Option[Int] = None
+
+  private[spark] var cachedArrowBatchServerSecret: Option[String] = None
```

Comment:
Shall we assign the values for driver too? Thus, driver and executor would have the consistent behaviour, e.g., both can access the vlaue through these local vars instead of the server itself.

Reply:
And, actually, I think we can create the `CachedArrowBatchServer` instance in `SparkEnv` so that driver and executor can reuse it.

Reply:
I am considering adding API like:

```
# 1. User calls this developer API in pyspark UDF
# to start a arrow stream server in local executor.
server_port, server_secret = startChunkServer()

# 2.read chunk data using the server created above.
# user can call this function in pyspark UDF or descendent processes
# of pyspark UDF.
readChunk(chunk_id, server_port, server_secret)

# 3. shut down the server created above
shutdownChunkServer(server_port, server_secret)
```

so that we can avoid each executor launches a long-running server.
https://docs.google.com/document/d/1qs8lKQ3IwF5QGGAaa6OIiXYhdG4_HJtS66dswtx9kd0/edit#bookmark=id.f6cwxc97g3ig

Then we don't need these variables